runtime/thunk/thunk.ts

```diff
@@ -367,7 +367,11 @@ function findParentFormula(
     case FormulaType.Sync:
     case FormulaType.SyncUpdate:
       if (syncTables) {
-        const syncTable = syncTables.find(table => table.getter.name === formulaSpec.parentFormulaName);
+        let syncTable = syncTables.find(table => table.getter.name === formulaSpec.parentFormulaName);
+        // Fallback to the name of the SyncTable itself.
```

Comment:
I'm happy to swap the priority of which one takes priority... but maybe we should just fix the fact that it matters? I just demonstrated that we allow a sync table's name to match another sync table's formula name, and that seems wrong:

<img width="572" height="712" alt="Image" src="https://github.com/user-attachments/assets/550d94f2-c7c7-48f2-8fe8-fb46816a2378" />